contracts/ScoutGameProtocol.sol

```diff
@@ -0,0 +1,85 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.26;
+
+import "@ethereum-attestation-service/eas-contracts/contracts/resolver/SchemaResolver.sol";
+import "@ethereum-attestation-service/eas-contracts/contracts/IEAS.sol";
+import "./ScoutToken.sol";
+
+/// @title ScoutGameProtocol
+/// @notice A schema resolver that manages unclaimed balances based on EAS attestations.
+contract ScoutGameProtocol is SchemaResolver {
+    ScoutToken public token;
+    mapping(address => uint256) private _unclaimedBalances;
+
+    // Define a constant for 18 decimals
+    uint256 constant DECIMALS = 10 ** 18;
+
+    constructor(IEAS eas, address _token) SchemaResolver(eas) {
+        token = ScoutToken(_token);
+    }
+
+    // Method that is called by the EAS contract when an attestation is made
+    //
+    function onAttest(
+        Attestation calldata attestation,
+        uint256 /*value*/
+    ) internal override returns (bool) {
+        uint256 addedBalance = 10 * DECIMALS;
```

Comment:
This would be a good place to add validations on who can attest

Example here
https://docs.attest.org/docs/tutorials/resolver-contracts#attester-resolver